code/game/objects/items/explosives/grenades/grenade.dm

```diff
@@ -83,7 +83,7 @@
 
 /obj/item/explosive/grenade/update_overlays()
 	. = ..()
-	if(active)
+	if(active && dangerous)
 		overlays += new /obj/effect/overlay/danger
```

Comment:
Oh, this is also wrong, although I know its old code. update overlays should just return a list of overlays it should currently have. So:
. += new /obj/effect/overlay/danger